pipeline/e2e-emulator-template.yaml

```diff
@@ -0,0 +1,36 @@
+parameters:
+  apiLevel: null
+
+jobs:
+  - job: 'e2e_test_API_${{ parameters.apiLevel }}'
+    dependsOn: 'gradlew_build'
+    pool:
+        vmImage: 'macOS-latest'
```

Comment:
Why macOS? Historically, I think the mac hosted agents have been both slower and flakier than Windows/Ubuntu agents

Reply:
Sadly, it is the only option [per ADO docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/ecosystems/android?view=azure-devops#test-on-the-android-emulator). Weirdly, the Android emulator is [installed on other agents](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md#android), but it didn't function properly when I tried with Ubuntu.

Reply:
That's reasonable! It might be good to leave a comment with a link to those docs explaining the decision

Reply:
Comment added!